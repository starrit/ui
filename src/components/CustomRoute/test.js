import React from 'react';
import {MemoryRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';

import {CustomRoute} from './';

function DummyComponent() {
  return <div>Auth protected Dummy Component</div>;
}

function DummyLoginComponent() {
  return <div>Login component</div>;
}

describe('CustomRoute Component', () => {
  it('should render component correctly', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <CustomRoute routes={[]} />
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render component with routes.js', () => {
    const routes = [{
      path: '/route1',
      component: DummyComponent,
      public: true
    }];

    const tree = renderer.create(
      <MemoryRouter initialEntries={['/route1']}>
        <CustomRoute routes={routes} />
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should redirect to Login route for non public route', () => {
    const routes = [{
      path: '/route1',
      component: DummyComponent
    }, {
      path: '/login',
      component: DummyLoginComponent,
      public: true
    }];

    const tree = renderer.create(
      <MemoryRouter initialEntries={['/route1']}>
        <CustomRoute routes={routes} />
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
