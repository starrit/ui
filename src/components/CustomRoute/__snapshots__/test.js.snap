// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomRoute Component should redirect to Login route for non public route 1`] = `
<div>
  Login component
</div>
`;

exports[`CustomRoute Component should render component correctly 1`] = `
<p>
  Page Not Found
</p>
`;

exports[`CustomRoute Component should render component with routes 1`] = `
<div>
  Auth protected Dummy Component
</div>
`;
